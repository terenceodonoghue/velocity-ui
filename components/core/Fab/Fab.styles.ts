import { css, FunctionInterpolation, Theme } from '@emotion/react';
import { darken } from 'polished';

const fab: FunctionInterpolation<Theme> = ({
  palette,
  transitions,
  typography,
}) => css`
  display: flex;
  align-items: center;
  justify-content: center;
  border: none;
  border-radius: 50%;
  height: 70px;
  padding: 0;
  width: 70px;
  background-color: ${palette.accent};
  color: ${palette.neutral[50]};
  font-weight: ${typography.fontWeight.medium};
  cursor: pointer;
  transition: background-color ${transitions.duration.standard}ms
    ${transitions.easing.easeInOut};

  &:disabled {
    opacity: 0.3;
    pointer-events: none;
  }

  &:focus {
    outline: 0;
  }

  &:enabled:hover {
    background-color: ${darken(palette.tonalOffset, palette.accent)};
  }
`;

export default fab;
