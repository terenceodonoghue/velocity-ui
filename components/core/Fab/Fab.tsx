import React, { ButtonHTMLAttributes, FunctionComponent } from 'react';
import fab from './Fab.styles';

const Fab: FunctionComponent<ButtonHTMLAttributes<HTMLButtonElement>> = (
  props,
) => <button css={fab} data-testid="fab" type="button" {...props} />;

export default Fab;
