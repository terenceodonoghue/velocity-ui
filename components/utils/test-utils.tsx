import React, { FunctionComponent, ReactElement } from 'react';
import { render, RenderOptions, RenderResult } from '@testing-library/react';
import ThemeProvider from '~/components/core/ThemeProvider';

const Providers: FunctionComponent = ({ children }) => {
  return <ThemeProvider>{children}</ThemeProvider>;
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'queries'>,
): RenderResult => render(ui, { wrapper: Providers, ...options });

export * from '@testing-library/react';
export { default as userEvent } from '@testing-library/user-event';

export { customRender as render };
