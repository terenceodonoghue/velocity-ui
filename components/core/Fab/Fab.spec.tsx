import React from 'react';
import { render } from 'test-utils';
import Fab from './Fab';

describe('Fab', () => {
  it('renders button', () => {
    const { container } = render(<Fab />);

    expect(container).toMatchSnapshot();
  });
});
