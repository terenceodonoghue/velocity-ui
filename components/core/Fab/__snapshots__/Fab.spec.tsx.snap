// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fab renders button 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  border: none;
  border-radius: 50%;
  height: 70px;
  padding: 0;
  width: 70px;
  background-color: #2e5bff;
  color: #fff;
  font-weight: 500;
  cursor: pointer;
  -webkit-transition: background-color 300ms cubic-bezier(0.4, 0, 0.2, 1);
  transition: background-color 300ms cubic-bezier(0.4, 0, 0.2, 1);
}

.emotion-0:disabled {
  opacity: 0.3;
  pointer-events: none;
}

.emotion-0:focus {
  outline: 0;
}

.emotion-0:enabled:hover {
  background-color: #002bc7;
}

<div>
  <button
    class="emotion-0"
    data-testid="fab"
    type="button"
  />
</div>
`;
